src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -97,21 +112,93 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
     toggle: togglePersonalisedSchedule,
   } = useShowHide(false);
 
+  const liveAndFutureEvents = useMemo(
+    () => relatedVenueEvents.filter(isEventLiveOrFuture),
+    [relatedVenueEvents]
+  );
+  const hasSavedEvents = liveAndFutureEvents
+    .map(
+      prepareForSchedule({
+        usersEvents: userEventIds,
+      })
+    )
+    .filter((event) => event.isSaved).length;
+
+  const isShowPersonalDownloadBtn = hasSavedEvents && showPersonalisedSchedule;
+  const liveEventsMinimalStartValue = Math.min(
+    ...liveAndFutureEvents.map((event) => event.start_utc_seconds)
+  );
+
+  const firstLiveEvent = liveAndFutureEvents.find(
+    (event) => event.start_utc_seconds === liveEventsMinimalStartValue
+  );
+
+  const minDate = useMemo(
+    () =>
+      firstLiveEvent
+        ? isEventLiveOrFuture(firstLiveEvent)
+          ? new Date(liveEventsMinimalStartValue).getTime()
+          : millisecondsToSeconds(todaysDateTime)
+        : minRangeValue,
+    [firstLiveEvent, liveEventsMinimalStartValue]
+  );
+
+  const firstRangeDayTimeValue = isDateLessOrEqualsToday({
+    dateValue: secondsToMilliseconds(minDate),
+    targetDateValue: millisecondsToSeconds(startOfToday().getTime()),
+  });
```

Comment:
`isDateLessOrEqualsToday` hints that the result of the function is `boolean` because of `is` in the beginning, though the const name is `firstRangeDayTimeValue` which does not sound like a boolean, please change one or the other.

Reply:
moreover, the name `isDateLessOrEqualsToday` does not sound like the function that takes 2 arguments.
for me the functionality of it, based on name is:
(date) => boolean
if date is later than today, then it's `true`, otherwise `false`.
although, we have some strange arguments like "targetDateValue". What is target here (based on the function name), not sure...

Reply:
renamed func + variable, so it's more explicit what it does 60ff2eb